relay.go

```diff
@@ -217,16 +217,15 @@ func (r *Relayer) handleCallReq(f lazyCallReq) error {
 	}
 	peer := r.peers.GetOrAdd(hostPort)
 
-	ctx, cancel := NewContext(time.Second)
-	defer cancel()
-
-	remoteConn, err := peer.GetConnection(ctx)
+	// TODO: Should connections use the call timeout? Or a separate timeout?
```

Comment:
Yeah, these really should be completely orthogonal. I think we want to get to:
 - separate connection management that limits the number of connection attempts we are making and has separate timeouts
 - in the call request path, if we need a connection, we pass how long we're willing to wait to get a connection

And in general, there's a lot of connection management (e.g., when should connections be closed)